src/ui/widgets/editors/w_ChainSha256Editor.cpp

```diff
@@ -0,0 +1,64 @@
+#include "w_ChainSha256Editor.hpp"
+
+#include "ui/widgets/common/WidgetUIBase.hpp"
+#include "utils/QvHelpers.hpp"
+
+#include <QDesktopServices>
+
+ChainSha256Editor::ChainSha256Editor(QWidget *parent, const QList<QString> &chain) : QDialog(parent)
+{
+    setupUi(this);
+    chainSha256Edit->setPlainText(chain.join("\r\n"));
+    QvMessageBusConnect(ChainSha256Editor);
+}
+
+QvMessageBusSlotImpl(ChainSha256Editor)
+{
+    switch (msg)
+    {
+        MBShowDefaultImpl;
+        MBHideDefaultImpl;
+        MBRetranslateDefaultImpl;
+        case UPDATE_COLORSCHEME: break;
+    }
+}
+
+void ChainSha256Editor::accept()
+{
+    const auto newChain = ChainSha256Editor::convertFromString(chainSha256Edit->toPlainText());
+    if (const auto err = ChainSha256Editor::validateError(newChain); err)
+    {
+        QvMessageBoxWarn(this, tr("Invalid Certificate Hash Chain"), *err);
+        return;
+    }
+    this->chain = newChain;
+    this->QDialog::accept();
+}
+
+QList<QString> ChainSha256Editor::convertFromString(const QString &&str)
+{
+    const static QRegExp newLine("[\r\n]");
```

Comment:
Will this cause issue on some platforms?
trimmed() may be better.

Reply:
`[\r\n]` will match `\r`, `\n`, for `\n\r` or `\r\n`, the flag `SkipEmptyParts` should help. so it doesn't matter.